packages/common/container-definitions/src/runtime.ts

```diff
@@ -260,4 +260,15 @@ export interface IGetPendingLocalStateProps {
 	 * be preserved and collected.
 	 */
 	readonly stopBlobAttachingSignal?: AbortSignal;
+
+	/**
+	 * Date to be used as the starting time of a session. This date is updated in case we refresh the
+	 * base snapshot since we won't be referencing ops older than the new snapshot.
+	 */
+	readonly sessionExpiryTimerStarted?: number;
+
+	/**
+	 * Snapshot sequence number. It will help the runtime to know which ops should still be stashed.
```

Comment:
We need to change narrative here. It's not about help, it's a contract - runtime has to satisfy that contract.
That said, to be honest - I do not understand what contract is.
The way it's written - runtime needs to ensure that it saves state that will be rehydrated on top of snapshot with this reference sequence number.
But I think runtime may not have a way to satisfy that requirement. So, I think negotiation should be in opposite direction - Loader layer should be asking runtime if it can use such snapshot (i.e. pull vs. push). Or maybe runtime continuously updating loader with the min seq number for base snapshot. Runtime would still need this info when generating stash.